### PR TITLE
[7.x] Cache::lock support for the database cache driver

### PR DESCRIPTION
**Pull request by taylorotwell**
This provides cache lock support for the database cache driver inspired by: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Lock/Store/PdoStore.php

Tests pending...

### CODE REVIEW

File: src/Illuminate/Cache/DatabaseLock.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+     * The database table name.
+     *
+     * @var string
+     */
+    protected $table;
+
+    /**
+     * The prune probability odds.
+     *
+     * @var string
```
Comment (marmichalski) [inline src/Illuminate/Cache/DatabaseLock.php:27]: You define it as string, but describe and default it to an array of ints 🤔

Comment (mfn) [reply src/Illuminate/Cache/DatabaseLock.php:27]: I wish Laravel lead devs would start thinking about a future with static analysers and more type declarations

Comment (driesvints) [reply src/Illuminate/Cache/DatabaseLock.php:27]: As soon as we have PHP 8 as the minimum version which has union type support we could *start* considering it. But that'd be September 2022 at the very earliest: https://www.php.net/supported-versions.php

File: src/Illuminate/Cache/DatabaseLock.php [hunk 2]
```diff
@@ -24,7 +24,7 @@ class DatabaseLock extends Lock
     /**
      * The prune probability odds.
      *
-     * @var string
+     * @var array
```
Comment (marmichalski) [inline src/Illuminate/Cache/DatabaseLock.php:27]: The same thing is in DatabaseStore. 🙄

**Pull request merged**